frontend/packages/metal3-plugin/src/components/host.tsx

```diff
@@ -1,57 +1,73 @@
+import * as _ from 'lodash-es';
+import * as classNames from 'classnames';
+import { sortable } from '@patternfly/react-table';
 
-import { getName, getNamespace } from '@console/shared';
-// TODO(jtomasek): update import once models are moved to console-shared package
-// import { MachineModel, NodeModel } from '@console/internal/models';
-import { referenceForModel } from '@console/internal/module/k8s';
+import { getName, getNamespace, getMachineNode } from '@console/shared';
+import { MachineModel, NodeModel } from '@console/internal/models';
 
-import {
-  ListHeader,
-  ColHead,
-  List,
-  ListPage,
-  ResourceRow,
-} from '@console/internal/components/factory';
-
-import { ResourceLink } from '@console/internal/components/utils';
-// import { WithResources } from '@console/shared';
+import { MultiListPage, Table, TableRow, TableData } from '@console/internal/components/factory';
+import { ResourceLink, Kebab } from '@console/internal/components/utils';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
 
 import { BaremetalHostModel } from '../models';
-import { getHostBMCAddress } from '../selectors';
+import { getHostBMCAddress, getHostMachine } from '../selectors';
+import { BaremetalHostRole } from './host-role';
 import MachineCell from './machine-cell';
 
-// const nameColumnClasses = 'col-lg-2 col-md-4 col-sm-6 col-xs-6';
-// const statusColumnClasses = 'col-lg-2 col-md-4 hidden-sm hidden-xs';
-// const machineColumnClasses = 'col-lg-3 visible-lg';
-// const roleColumnClasses = 'col-lg-2 visible-lg';
-// const addressColumnClasses = 'col-lg-2 visible-lg';
-const columnClasses = 'col-sm-4';
-
-const HostHeader = (props: React.ComponentProps<typeof ColHead>) => (
-  <ListHeader>
-    <ColHead {...props} className={columnClasses} sortField="metadata.name">
-      Name
-    </ColHead>
-    {/* <ColHead {...props} className={statusColumnClasses}>
-      Status
-    </ColHead> */}
-    <ColHead {...props} className={columnClasses}>
-      Machine
-    </ColHead>
-    {/* <ColHead {...props} className={roleColumnClasses}>
-      Role
-    </ColHead> */}
-    <ColHead {...props} className={columnClasses} sortField="spec.bmc.address">
-      Management Address
-    </ColHead>
-  </ListHeader>
-);
+const tableColumnClasses = [
+  classNames('col-lg-5', 'col-md-8', 'col-sm-12', 'col-xs-12'),
+  classNames('col-lg-2', 'col-md-4', 'col-sm-6', 'hidden-xs'),
+  classNames('col-lg-3', 'col-md-4', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-2', 'hidden-md', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-2', 'hidden-md', 'hidden-sm', 'hidden-xs'),
+  Kebab.columnClass,
+];
+
+const HostsTableHeader = () => [
+  {
+    title: 'Name',
+    sortField: 'metadata.name',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[0] },
+  },
+  {
+    title: 'Machine',
+    sortField: 'spec.machineRef.name',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[2] },
```

Comment:
We're skipping a column class. Is that intentional?

Reply:
Yes, that one is going to be used by `Host Status`. I've updated first column to span its space for the time being.